modules/ln_glgui/primitives.scm

```diff
@@ -61,11 +61,13 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
     (glCoreEnd)))
 
 (define (glgui:draw-linestrip data color . width)
-  (if (and (list? data) (fx> (length data) 1) (fx< (length data) 200)) (begin
-    (let loop ((pts data))
-      (if (< (length pts) 2) #t (begin
-        (apply glgui:draw-line (append (car pts) (cadr pts) (list color) width))
-        (loop (cdr pts))))))))
+  (if (and (pair? data) (fx< (length data) 200)) ;; FIXME: pass the length in as parameter
```

Comment:
This future todo may need to be done in a backwards compatible way in case people have private widgets that use rely on the current argument order and this being done internally?